CODINGSTANDARDS.md

```diff
@@ -0,0 +1,241 @@
+# Coding conventions for crocks
+
+## tl;dr;
+
+* `npm run setup` installs all the things
+* `npm run lint` to check for styling issues
+* `npm run spec:dev` to have tests running while coding
+* `npm run spec:coverage` to ensure coverage is within expected levels
+
+## Coding Standards
+
+This page will act as the basis for the `crocks` coding standards. The more
+this document is followed, the less comments you will receive on your PR. If you
+find anything that does not match these coding standards, feel free to create a
+PR to align the code with them
+
+### Header
+
+Each main file (not `.spec`) file has a header similiar to the below. If you add
+a new file, put your own details in as well as the current year. If the file
+already has a header there is no need to update it.
+
+```javascript
+/** @license ISC License (c) copyright 2019 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+```
+
+### Main File Structure
+
+The following is an example of the normal structure of a core library file 
+using `find` as an example of this structure. It is also a good example of
+how each argument should be validated.
+
+#### Header
+
+As stated above, the header should display the year of authorship along with
+the details of the author within the structure of the following license statement
+
+```javascript
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+```
+
+#### Imports
+
+The imports should be in two alphabetized groups with monoids as the first
+group and functions as the second.
+
+```javascript
+const Pred = require('../core/types').proxy('Pred')
+
+const curry = require('../core/curry')
+const predOrFunc = require('../core/predOrFunc')
+const isFunction = require('../core/isFunction')
+const isFoldable = require('../core/isFoldable')
+const isSameType = require('../core/isSameType')
+```
+
+#### Private functions
+
+Private functions or destructing functions from your imports are done between
+the imports and the main function.
+
+```javascript
+const { Just, Nothing } = require('.')
+
+const accumulator = fn => (acc, cur) =>
+  !acc.found && predOrFunc(fn, cur) ? { found: true, value: cur } : acc
+```
+
+#### Main function
+
+The main function must have a function signature written above using the haskell
```

Comment:
`haskell arrow notation` -> `Hindley–Milner type notation`

Reply:
THATS THE NAME! omg i searched everywhere for that and all i kept getting was `haskell notation`